Desktop/QueryRetrieve SCU/Program.cs

```diff
@@ -0,0 +1,239 @@
+            var cMoveRequest = CreateCMoveBySeriesUID("FODICOMSCP", studyUID, serieUids.First());
+            bool? moveSuccessfully = null;
+            cMoveRequest.OnResponseReceived += (DicomCMoveRequest requ, DicomCMoveResponse response) =>
+            {
+                if (response.Status.State == DicomState.Pending)
+                {
+                    Console.WriteLine("Sending is in progress. please wait");
+                }
+                else if (response.Status.State == DicomState.Success)
+                {
+                    Console.WriteLine("Sending successfully finished");
+                    moveSuccessfully = true;
+                }
+                else if (response.Status.State == DicomState.Failure)
+                {
+                    Console.WriteLine("Error sending datasets: " + response.Status.Description);
+                    moveSuccessfully = false;
+                }
+                Console.WriteLine(response.Status);
+            };
+            client.AddRequest(cMoveRequest);
+            client.Send(QRServerHost, QRServerPort, false, AET, QRServerAET);
+
+            if (moveSuccessfully.GetValueOrDefault(false))
+            {
+                // images sent successfully from QR Server to the store scp
+            }
+            Console.ReadLine();
+        }
+
+
+        public static DicomCFindRequest CreateStudyRequestByPatientName(string patientName)
+        {
+            var request = new DicomCFindRequest(DicomQueryRetrieveLevel.Study);
+
+            request.Dataset.AddOrUpdate(new DicomTag(0x8, 0x5), "ISO_IR 100");
+
+            // add the dicom tags with empty values that should be included in the result
+            request.Dataset.AddOrUpdate(DicomTag.StudyDate, "");
+            request.Dataset.AddOrUpdate(DicomTag.StudyTime, "");
+            request.Dataset.AddOrUpdate(DicomTag.AccessionNumber, "");
+            request.Dataset.AddOrUpdate(DicomTag.ModalitiesInStudy, "");
+            request.Dataset.AddOrUpdate(DicomTag.StudyDescription, "");
+            request.Dataset.AddOrUpdate(DicomTag.InstitutionName, "");
+            request.Dataset.AddOrUpdate(DicomTag.ReferringPhysicianName, "");
+            request.Dataset.AddOrUpdate(DicomTag.ProtocolName, "");
+            request.Dataset.AddOrUpdate(DicomTag.PatientName, "");
+            request.Dataset.AddOrUpdate(DicomTag.PatientID, "");
+            request.Dataset.AddOrUpdate(DicomTag.PatientBirthDate, "");
+            request.Dataset.AddOrUpdate(DicomTag.PatientSex, "");
+            request.Dataset.AddOrUpdate(DicomTag.StudyInstanceUID, "");
+            request.Dataset.AddOrUpdate(DicomTag.StudyID, "");
+
+            // add the dicom tags that contain the filter criterias
+            request.Dataset.AddOrUpdate(DicomTag.PatientName, patientName);
+
+            return request;
+        }
+
+
+        public static DicomCFindRequest CreateSeriesRequestByStudyUID(string studyInstanceUID)
```

Comment:
Why aren't you using the library method `DicomCFindRequest.CreateSeriesQuery` for this purpose?